### PR TITLE
Add ADR about time graph filtering TSP updates

### PR DESCRIPTION
**Pull request by hriday-panchasara**
Signed-off-by: hriday-panchasara <hriday.panchasara@ericsson.com>

### CODE REVIEW

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:54]: You're mentioning Trace Compass. Please explain it from the theia-trace-extension / TSP point of view.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:53]: Adjust the description based on the new name.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:59]: This might not be the final UI, but it gives the idea how to achieve it.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:60]: change full-search

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:65]: `strategy` instead of `is_full_search`.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:114]: The reason for a bit mask is to minimize the amount of data transported per state. In the front-end you can apply a style for each state.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:116]: Can add a comment about the filter DSL? Mabye under future enhancements. Basically, the first implementation will pass the string to the back-end and it will apply it. There is no feedback if the syntax was correct. 

However, there is a possibility to create LSP server (Language Server Protocol) in the back-end and front-end will use it to do syntax validation. Trace Compass incubator already has such a server that we can integrate and have an example implementation.

**Review by bhufmann — CHANGES_REQUESTED** [review]

File: doc/adr/0009-time-graph-search-functionality.md
Comment (hriday-panchasara) [reply doc/adr/0009-time-graph-search-functionality.md]: Done

File: doc/adr/0009-time-graph-search-functionality.md
Comment (hriday-panchasara) [reply doc/adr/0009-time-graph-search-functionality.md]: Done

File: doc/adr/0009-time-graph-search-functionality.md
Comment (hriday-panchasara) [reply doc/adr/0009-time-graph-search-functionality.md]: Done

File: doc/adr/0009-time-graph-search-functionality.md
Comment (hriday-panchasara) [reply doc/adr/0009-time-graph-search-functionality.md:76]: Done

File: doc/adr/0009-time-graph-search-functionality.md
Comment (hriday-panchasara) [reply doc/adr/0009-time-graph-search-functionality.md]: Done

File: doc/adr/0009-time-graph-search-functionality.md
Comment (hriday-panchasara) [reply doc/adr/0009-time-graph-search-functionality.md]: Done

File: doc/adr/0009-time-graph-search-functionality.md
Comment (hriday-panchasara) [reply doc/adr/0009-time-graph-search-functionality.md:137]: Done

**Comment by hriday-panchasara** [comment]
Added a sequence diagram to show the order of http operations to make the filter work. This diagram is pretty basic, I was not sure if I needed to go in depth.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:140]: Here you should propose which solution to implement and the way to get there. Maybe describe increments if applicable.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:146]: Maybe describe in short the main benefit for the end-user.

**Review by bhufmann — CHANGES_REQUESTED** [review]
Small additional comments from my side. It would be good that other committers would talke part of the review before I approve it.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (Rodrigoplp-work) [inline doc/adr/0009-time-graph-search-functionality.md:31]: Would it be "- The array `requested_timerange`" instead of `requested_times`?

File: doc/adr/0009-time-graph-search-functionality.md
Comment (Rodrigoplp-work) [inline doc/adr/0009-time-graph-search-functionality.md:75]: This format works, but we tend to prefer listing links at the end of the document, below the last line of text (see ADR 0002-GraphQL for an example).

So here would be:
`- [https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886][pull-886]`

And at the end of the ADR:
`[pull-886]: https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886`

File: doc/adr/0009-time-graph-search-functionality.md
Comment (Rodrigoplp-work) [inline doc/adr/0009-time-graph-search-functionality.md:109]: Move links to the bottom of the ADR.

**Review by Rodrigoplp-work — CHANGES_REQUESTED** [review]
Excellent contribution! I have just a few minor requests intended to keep the format of the ADR a bit more in line with the previous ones.

**Review by bhufmann — APPROVED** [review]
LGTM. Thanks for the proposal. Looking forward seeing that in action.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [inline doc/adr/0009-time-graph-search-functionality.md:7]: @hriday-panchasara  We should change it to Accepted.

**Review by Rodrigoplp-work — DISMISSED** [review]

**Review by bhufmann — APPROVED** [review]
Renewed approval! Thanks.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (frallax) [inline doc/adr/0009-time-graph-search-functionality.md:97]: Does anyone think that this proposal should also consider / compare to the possibility of using graphQL to make such "filtering queries"? I am not sure if [adr 002] was meant to look into this or not, but I think that at least a few words regarding if graphQL would be adapt to do similar "filter queries" or not could be added

[adr 002]: https://github.com/eclipse-cdt-cloud/theia-trace-extension/blob/master/doc/adr/0002-GraphQL.md

File: doc/adr/0009-time-graph-search-functionality.md
Comment (frallax) [inline doc/adr/0009-time-graph-search-functionality.md:140]: is there an action point / issue (to link here) to follow up on this activity? Just to make sure that this "time-graph filtering feature" will see the light after the ADR is merged, and does not get forgotten / paused

**Review by MatthewKhouzam — APPROVED** [review]
I like this. But I want this to be a simple +1, not a merge now +1

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [reply doc/adr/0009-time-graph-search-functionality.md:97]: No, I don't think GraphQL should be considered for "filtering queries". ADR 002 investigated GraphQL as replacement of the current TSP implementation.

File: doc/adr/0009-time-graph-search-functionality.md
Comment (bhufmann) [reply doc/adr/0009-time-graph-search-functionality.md:140]: After re-reading the statement, I realized that it needs to be re-phrased. The decision is to go with this suggestion and NOT to start a new discussion on how the TSP should look like.

Comment (bhufmann) [reply doc/adr/0009-time-graph-search-functionality.md:140]: @hriday-panchasara could you please re-phrase that sentence? The decision is to use `filter_query_parameters`, `strategy` to add search/filter expressions to the `/states` query that will return `tags` bit mask.

**Review by bhufmann — CHANGES_REQUESTED** [review]
Sorry for the late request, please update the decision section. This ADR is meant to decide on the API  and the decision section should reflect that.

**Review by Rodrigoplp-work — APPROVED** [review]

**Pull request merged**
